river/naive_bayes/bernoulli.py

```diff
@@ -136,3 +217,66 @@ def joint_log_likelihood(self, x):
             )
             for c in self.class_counts
         }
+
+    def learn_many(self, X: pd.DataFrame, y: pd.Series):
+        X = (X > self.true_threshold) * 1
+
+        agg, index = base.Groupby(keys=y).apply(np.sum, X.values)
```

Comment:
Could you try comparing with a for loop over the values of `y`? So something like:

```py
for y_val in y.unique():
    counts = X[y == y_val].sum()
    # do your update
``` 

I don't know, maybe your approach is better. I'm just wondering if there isn't a much faster way to do this that we aren't seeing :)

Reply:
It seems that the GroupBy class is more efficient on my benchmark, up to 3 times faster than a for loop. Agree with you, we need to benchmark things. 🙂

```python
def learn_many(self, X: pd.DataFrame, y: pd.Series):
    for y_val in y.unique():
        agg = X[y == y_val].sum()
        self.class_counts.update({y_val: y[y == y_val].sum()})
        self.feature_counts.update((agg.T).to_dict())
        self.class_totals.update({y_val: agg.sum()})

    return self

X_t = feature_extraction.BagOfWords().transform_many(X)

model = naive_bayes.MultinomialNB()

%timeit learn_many(self=model, X=X_t, y=Y)
269 ms ± 46 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

```python
 model = naive_bayes.MultinomialNB()

%timeit model.learn_many(X_t, Y)
73.8 ms ± 15.9 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```